src/toil/batchSystems/mesos/batchSystem.py

```diff
@@ -352,8 +352,9 @@ def resourceOffers(self, driver, offers):
 
         # Right now, gives priority to largest jobs
         for offer in offers:
-            tasks = []
             # TODO: In an offer, can there ever be more than one resource with the same name?
+            allTasks = []
+            subTasks = defaultdict(list)
```

Comment:
Very true. I used a dict just in case we decided to use the info in the data structure for some logging or other operations in the future